Modules/_ssl.c

```diff
@@ -4399,8 +4400,10 @@ _ssl__SSLContext_set_ecdh_curve(PySSLContext *self, PyObject *name)
     SSL_CTX_set_tmp_ecdh(self->ctx, key);
     EC_KEY_free(key);
 #else
-    if (!SSL_CTX_set1_groups(self->ctx, &nid, 1)) {
-        _setSSLError(get_state_ctx(self), NULL, 0, __FILE__, __LINE__);
+    int res = SSL_CTX_set1_groups_list(self->ctx, PyBytes_AS_STRING(name_bytes));
```

Comment:
SSL_CTX_set1_groups_list also supports another syntax, adding a `?` before the curve name makes it "optional", [quoting the docs](https://www.openssl.org/docs/manmaster/man3/SSL_CTX_set1_groups_list.html):

> If a group name is preceded with the ? character, it will be ignored if an implementation is missing.

This isn't tested in our test suite, it would be useful to test that so future contributors know that syntax exists.